File: LeetCode/pow.c
#include<stdio.h>

int main() {
    int x = 2;
    int pow = 7;
    
    int result = x;
    while (pow > 1) {
        if (pow % 2 == 0) {
            result = result * x * x;
        } else {
            result = result * x * x * x;
        }
        pow /= 2;
    }
    
    printf("%d \n", result);
    return 1;
}

File: LeetCode/pow.cs
using System;

class Program {
    static void Main() {
        int x = 2;
        int n = 5;
        int pow = n;
        int result = 1;
        while (n > 1) {
            if (n % 2 == 0) {
                result *= x * x;
            } else {
                result *= x * x * x;
            }
            n /= 2;
        }

        Console.WriteLine("{0} ^ {1} = {2}", x, pow, result);
    }
}
